tests/functional/modules/test_kubernetesmod.py

```diff
@@ -0,0 +1,1086 @@
+import logging
+
+import pytest
+from salt.exceptions import CommandExecutionError
+from saltfactories.utils import random_string
+
+log = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.skip_unless_on_linux(reason="Only run on Linux platforms"),
+]
+
+
+@pytest.fixture
+def kubernetes(modules):
+    """
+    Return the kubernetes execution module
+    """
+    return modules.kubernetes
+
+
+@pytest.fixture
+def namespace_name():
+    """
+    Fixture providing a namespace name for testing
+
+    Kubernetes requires names to be lowercase RFC 1123 complient
+    """
+    return random_string("namespace-", uppercase=False)
+
+
+@pytest.fixture(params=[True])
+def namespace(kubernetes, namespace_name, request):
+    """
+    Fixture to create a test namespace.
+    """
+    name = namespace_name
```

Comment:
praise: The new tests look awesome!
small suggestion: I don't think we need to separate name generation from the fixture itself (valid for all `*_name`):

```suggestion
@pytest.fixture(params=[True])
def namespace(kubernetes, namespace_name, request):
    """
    Fixture to create a test namespace.
    """
    # Kubernetes requires names to be lowercase RFC 1123 compliant
    name = random_string("namespace-", uppercase=False)
```

Reply:
I have to leave node_name as a separate fixture due to the way it is generating the name. If I try to put it in the node_label fixture I get a pylint error `tests/functional/states/test_kubernetes.py:1229:16: R1708: Do not raise StopIteration in generator, use return statement instead (stop-iteration-return)` or I need to change how that is done.

Reply:
Ah, that's totally fine in that case. Since the node name doesn't change afaict, you can make it `scope="module"` though.